google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/strategy/CreateNodesWithGroupEncodedIntoNameThenAddToSet.java

```diff
@@ -103,14 +103,15 @@
 
       String sharedResourceName = namingConvention.create().sharedNameForGroup(group);
       Template mutableTemplate = template.clone();
-      GoogleComputeEngineTemplateOptions templateOptions = GoogleComputeEngineTemplateOptions.class
-            .cast(mutableTemplate.getOptions());
+      GoogleComputeEngineTemplateOptions templateOptions = GoogleComputeEngineTemplateOptions.class.cast(mutableTemplate.getOptions());
       assert template.getOptions().equals(templateOptions) : "options didn't clone properly";
 
-      // get or insert the network and insert a firewall with the users
-      // configuration
+      // get or insert the network and insert a firewall with the users configuration
       Network network = getOrCreateNetwork(templateOptions, sharedResourceName);
-      getOrCreateFirewalls(templateOptions, network, firewallTagNamingConvention.get(group));
+      // add firewalls *only* if the template options don't contain explicit network
+      if (templateOptions.network() == null) {
```

Comment:
I think that would be an inconvenient way to address whatever this PR tries to fix. If the intention is to avoid having duplicate firewall rules or semantically equivalent firewall rules, then the code that creates the rules should be fixed to avoid duplicating them.

Not creating the firewalls when the user explicitly calls the `inboundPorts` method should not be an option. We need to honor the portable interface and properly implement it. It is not OK to have *some* providers ignoring the interface contract *sometimes*. That reduces code portability a lot and increases considerably the complexity of the portable layer. It does not make sense to have a portable layer that has a different contract for each provider.